File: backend/app/schemas/trip.py
from datetime import datetime
from enum import Enum

from pydantic import BaseModel


class PointSource(str, Enum):
    gnss = "gnss"
    fused = "fused"
    dr_only = "dr_only"


class TrajectoryPointIn(BaseModel):
    ts: datetime
    lat: float
    lon: float
    source: PointSource
    accuracy_m: float | None = None
    covariance_major_m: float | None = None
    covariance_minor_m: float | None = None
    heading_deg: float | None = None


class TripOut(BaseModel):
    id: str
    user_id: str
    started_at: datetime
    ended_at: datetime | None = None
    distance_m: float | None = None
    max_drift_m: float | None = None
    pct_time_dr: float | None = None

File: mobile/lib/features/trips/domain/trip.dart
enum PointSource { gnss, fused, drOnly }

class Trip {
  const Trip({
    required this.id,
    required this.startedAt,
    this.endedAt,
    this.distanceM,
    this.maxDriftM,
    this.pctTimeDr,
  });

  final String id;
  final DateTime startedAt;
  final DateTime? endedAt;
  final double? distanceM;
  final double? maxDriftM;
  final double? pctTimeDr;
}

class TrajectoryPoint {
  const TrajectoryPoint({
    required this.ts,
    required this.lat,
    required this.lon,
    required this.source,
    this.accuracyM,
    this.covarianceMajorM,
    this.covarianceMinorM,
    this.headingDeg,
  });

  final DateTime ts;
  final double lat;
  final double lon;
  final PointSource source;
  final double? accuracyM;
  final double? covarianceMajorM;
  final double? covarianceMinorM;
  final double? headingDeg;
}
